### PR TITLE
Update office-365-isolation-in-office-365.md

### PR DESCRIPTION
**Pull request by msbemba**
Added Data storage location to the document.

### CODE REVIEW

**Comment by msbemba** [comment]
Issue#https://github.com/MicrosoftDocs/OfficeDocs-Enterprise/issues/863

**Comment by mypil** [comment]
@JohanFreelancer9 - kindly do the copy editor review for this PR. Thanks.

File: Enterprise/office-365-isolation-in-office-365.md [hunk 1]
```diff
@@ -65,3 +65,33 @@ A unique *SubscriptionId* is used for each tenant. All customer sites are owned
...
 
 SharePoint Online stores encrypted file content in Microsoft Azure blobs. Each SharePoint Online farm has its own Microsoft Azure account and all the blobs saved in Azure are encrypted individually with a key stored in the SQL content store. The encryption key protected in code by the authorization layer and not exposed directly to the end user. SharePoint Online has real-time monitoring to detect when an HTTP request reads or writes data for more than one tenant. The request identity *SubscriptionId* is tracked against the *SubscriptionId* of the accessed resource. Requests to access resources of more than one tenant should never happen by end users. Service requests in a multi-tenant environment are the only exception. For example, the search crawler pulls content changes for an entire database all at once. This usually involves querying sites of more than one tenant in a single service request, which is done for efficiency reasons.
+
+## Teams
+
+Your Teams data is stored differently depending on the content type.
```
Comment (JohanFreelancer9) [inline Enterprise/office-365-isolation-in-office-365.md:71]:
```suggestion
Your Teams data is stored differently, depending on the content type. 
```

File: Enterprise/office-365-isolation-in-office-365.md [hunk 2]
```diff
@@ -65,3 +65,33 @@ A unique *SubscriptionId* is used for each tenant. All customer sites are owned
...
+
+Your Teams data is stored differently depending on the content type. 
+
+Check out the [Ignite breakout session on Microsoft Teams architecture](https://channel9.msdn.com/Events/Ignite/Microsoft-Ignite-Orlando-2017/BRK3071) for an in-depth discussion.
+
+### Core Teams customer data
+
+If your tenant is provisioned in Australia, Canada, the European Union, France, Germany, India, Japan, South Africa, South Korea, Switzerland (which includes Liechtenstein), the United Arab Emirates, the United Kingdom, or the United States, Microsoft stores the following customer data at rest only within that location:
+
+- Teams chats, team and channel conversations, images, voicemail messages, and contacts
```
Comment (JohanFreelancer9) [inline Enterprise/office-365-isolation-in-office-365.md:79]:
```suggestion
- Teams chats, team and channel conversations, images, voicemail messages, and contacts.
```

File: Enterprise/office-365-isolation-in-office-365.md [hunk 3]
```diff
@@ -65,3 +65,33 @@ A unique *SubscriptionId* is used for each tenant. All customer sites are owned
...
+- SharePoint Online site content and the files stored within that site
```
Comment (JohanFreelancer9) [inline Enterprise/office-365-isolation-in-office-365.md:80]:
```suggestion
- SharePoint Online site content and the files stored within that site.
```

File: Enterprise/office-365-isolation-in-office-365.md [hunk 4]
```diff
@@ -65,3 +65,33 @@ A unique *SubscriptionId* is used for each tenant. All customer sites are owned
...
+- Files uploaded to OneDrive for Business
```
Comment (JohanFreelancer9) [inline Enterprise/office-365-isolation-in-office-365.md:81]:
```suggestion
- Files uploaded to OneDrive for work or school.
```

File: Enterprise/office-365-isolation-in-office-365.md [hunk 5]
```diff
@@ -65,3 +65,33 @@ A unique *SubscriptionId* is used for each tenant. All customer sites are owned
...
+
+#### Chat, channel messages, team structure
+
+Every team in Teams is backed by a Microsoft 365 Group and its SharePoint site and Exchange mailbox. Private chats (including group chats), messages sent as part of a conversation in a channel, and the structure of teams and channels are stored in a chat service running in Azure. The data is also stored in a hidden folder in the user and group mailboxes to enable Information Protection features.
+
+#### Voicemail and contacts
+
+Voicemails are stored in Exchange. Contacts are stored in Exchange-based cloud data store. Exchange and the Exchange-based cloud store already provide data residency in each of the worldwide datacenter geos. For all teams, voicemail and contacts are stored in-country for Australia, Canada, France, Germany, India, Japan, the United Arab Emirates, the United Kingdom, South Africa, South Korea, Switzerland (includes Liechtenstein), and the United States. For all other countries, files are stored in the US, Europe, or Asia-Pacific location based on tenant affinity.
```
Comment (JohanFreelancer9) [inline Enterprise/office-365-isolation-in-office-365.md:89]:
```suggestion
Voicemails are stored in Exchange. Contacts are stored in Exchange-based cloud data store. Exchange and the Exchange-based cloud store already provide data residency in each of the worldwide datacenter geos. For all teams, voicemail and contacts are stored in-country for Australia, Canada, France, Germany, India, Japan, the United Arab Emirates, the United Kingdom, South Africa, South Korea, Switzerland (which includes Liechtenstein), and the United States. For all other countries, files are stored in the US, Europe, or Asia-Pacific location based on tenant affinity.
```

File: Enterprise/office-365-isolation-in-office-365.md [hunk 6]
```diff
@@ -65,3 +65,33 @@ A unique *SubscriptionId* is used for each tenant. All customer sites are owned
...
+
+#### Images and media
+
+Media used in chats (except for Giphy GIFs which aren't stored but are a reference link to the original Giphy service URL, Giphy is a non-Microsoft service) is stored in an Azure-based media service that is deployed to the same locations as the chat service.
+
+#### Files
+
+Files (including OneNote and Wiki) that somebody shares in a channel are stored in the team’s SharePoint site. Files shared in a private chat or a chat during a meeting or call are uploaded and stored in the OneDrive for the Business account of the user who shares the file. Exchange, SharePoint and OneDrive already provide data residency in each of the worldwide datacenter geos. So, for existing customers, all files, OneNote notebooks, Teams wiki content, and mailboxes that are part of the Teams experience are already stored in the location based on your tenant affinity. Files are stored in-country for Australia, Canada, France, Germany, India, Japan, the United Arab Emirates, the United Kingdom, South Africa, South Korea, and Switzerland (which includes Liechtenstein). For all other countries, files are stored in the US, Europe, or Asia Pacific location based on tenant affinity.
```
Comment (JohanFreelancer9) [inline Enterprise/office-365-isolation-in-office-365.md:97]:
```suggestion
Files (including OneNote and Wiki) that somebody shares in a channel are stored in the team’s SharePoint site. Files shared in a private chat or a chat during a meeting or call are uploaded and stored in the OneDrive for work or school account of the user who shares the file. Exchange, SharePoint, and OneDrive already provide data residency in each of the worldwide datacenter geos. So, for existing customers, all files, OneNote notebooks, Teams wiki content, and mailboxes that are part of the Teams experience are already stored in the location based on your tenant affinity. Files are stored in-country for Australia, Canada, France, Germany, India, Japan, the United Arab Emirates, the United Kingdom, South Africa, South Korea, and Switzerland (which includes Liechtenstein). For all other countries, files are stored in the US, Europe, or Asia Pacific location based on tenant affinity.
```

Comment (JohanFreelancer9) [reply Enterprise/office-365-isolation-in-office-365.md:97]: Please change "OneDrive for the Business" to "OneDrive for work or school".

File: Enterprise/office-365-isolation-in-office-365.md (hunk 5 above)
Comment (JohanFreelancer9) [reply Enterprise/office-365-isolation-in-office-365.md:89]: Please change "(includes Liechtenstein)" to "(which includes Liechtenstein)".

**Comment by mypil** [comment]
@robmazz - PR has been copy edited and is ready for final review. This is based on issue #863.

Thank you.

**Comment by robmazz** [comment]
@mypil - I'm going to coordinate with a MS Teams contact on the review of these suggested changes. Thanks for the reminder!

**Comment by robmazz** [comment]
@mypil - confirmed with the Teams team and all this was taken from another article (https://docs.microsoft.com/en-us/microsoftteams/location-of-data-in-teams). I'll accept and merge for now, but may rework so as to not duplicate content in two different articles.

**Comment by Larryngolog** [comment]
Hello, sorry for  late reply, I have missed the notification about the comment. Provided informaiton just documents where Teams stores data but does not say anything about data isolation for pure Teams data - chat data and meetings. Is it possible to add this information to this documentation?

**Pull request merged**